src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Services/AzureSqlManagedInstanceAdapter.cs

```diff
@@ -0,0 +1,209 @@
+            Communicator = new AzureSqlManagedInstanceCommunicator(Context);
+        }
+
+        /// <summary>
+        /// Gets a managed instance in a resource group
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <param name="managedInstanceName">The name of the managed instance</param>
+        /// <returns>The managed instance</returns>
+        public AzureSqlManagedInstanceModel GetManagedInstance(string resourceGroupName, string managedInstanceName)
+        {
+            var resp = Communicator.Get(resourceGroupName, managedInstanceName);
+            return CreateManagedInstanceModelFromResponse(resp);
+        }
+
+        /// <summary>
+        /// Gets a list of all the managed instances in a subscription
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <returns>A list of all the managed instances</returns>
+        public List<AzureSqlManagedInstanceModel> ListManagedInstances()
+        {
+            var resp = Communicator.List();
+            return resp.Select((s) =>
+            {
+                return CreateManagedInstanceModelFromResponse(s);
+            }).ToList();
+        }
+
+        /// <summary>
+        /// Gets a list of all the managed instances in a resource group
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <returns>A list of all the managed instances</returns>
+        public List<AzureSqlManagedInstanceModel> ListManagedInstancesByResourceGroup(string resourceGroupName)
+        {
+            var resp = Communicator.ListByResourceGroup(resourceGroupName);
+            return resp.Select((s) =>
+            {
+                return CreateManagedInstanceModelFromResponse(s);
+            }).ToList();
+        }
+
+        /// <summary>
+        /// Upserts a managed instance
+        /// </summary>
+        /// <param name="model">The managed instance to upsert</param>
+        /// <returns>The updated managed instance model</returns>
+        public AzureSqlManagedInstanceModel UpsertManagedInstance(AzureSqlManagedInstanceModel model)
+        {
+            var resp = Communicator.CreateOrUpdate(model.ResourceGroupName, model.FullyQualifiedDomainName, new Management.Sql.Models.ManagedInstance()
+            {
+                Location = model.Location,
+                Tags = model.Tags,
+                AdministratorLogin = model.AdministratorLogin,
+                AdministratorLoginPassword = model.AdministratorPassword != null ? Decrypt(model.AdministratorPassword) : null,
+                Sku = model.Sku != null ? new Management.Sql.Models.Sku(model.Sku.Name) : null,
+                LicenseType = model.LicenseType,
+                StorageSizeInGB = model.StorageSizeInGB,
+                SubnetId = model.SubnetId,
+                VCores = model.VCores,
```

Comment:
oh, the API has `Vcores`? I see. very interesting. For DB/pool, this is `sku.capacity`. No need to change this now, just FYI.

Reply:
Ok